### PR TITLE
[AMORO-2192] Fix S3 related configurations not working

### LINKED ISSUE #2192: [Bug]: The S3 related configurations not working when setting S3 as storage type
### What happened?

When add an Internal Catalog working with S3, the configurations(like region) not working.

### Affects Versions

0.6.0

### What engines are you seeing the problem on?

Core, AMS

### How to reproduce

Create an catalog like
image

Execute SQL
image

### Relevant log output

```shell
2023/10/26 15:24:54 org.apache.iceberg.exceptions.ServiceFailureException: Server error: SdkClientException: Unable to load region from any of the providers in the chain software.amazon.awssdk.regions.providers.DefaultAwsRegionProviderChain@761f92c3: [software.amazon.awssdk.regions.providers.SystemSettingsRegionProvider@16b17512: Unable to load region from system settings. Region must be specified either via environment variable (AWS_REGION) or  system property (aws.region)., software.amazon.awssdk.regions.providers.AwsProfileRegionProvider@2270b3f9: No region provided in profile: default, software.amazon.awssdk.regions.providers.InstanceProfileRegionProvider@7245fbf4: Unable to contact EC2 metadata service.]
	at org.apache.iceberg.rest.ErrorHandlers$DefaultErrorHandler.accept(ErrorHandlers.java:162)
	at org.apache.iceberg.rest.ErrorHandlers$TableErrorHandler.accept(ErrorHandlers.java:109)
	at org.apache.iceberg.rest.ErrorHandlers$TableErrorHandler.accept(ErrorHandlers.java:93)
	at org.apache.iceberg.rest.HTTPClient.throwFailure(HTTPClient.java:176)
	at org.apache.iceberg.rest.HTTPClient.execute(HTTPClient.java:285)
	at org.apache.iceberg.rest.HTTPClient.execute(HTTPClient.java:219)
	at org.apache.iceberg.rest.HTTPClient.post(HTTPClient.java:330)
	at org.apache.iceberg.rest.RESTClient.post(RESTClient.java:112)
	at org.apache.iceberg.rest.RESTSessionCatalog$Builder.create(RESTSessionCatalog.java:603)
	at org.apache.iceberg.CachingCatalog$CachingTableBuilder.lambda$create$0(CachingCatalog.java:261)
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.lambda$doComputeIfAbsent$14(BoundedLocalCache.java:2344)
	at java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1853)
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.doComputeIfAbsent(BoundedLocalCache.java:2342)
	at com.github.benmanes.caffeine.cache.BoundedLocalCache.computeIfAbsent(BoundedLocalCache.java:2325)
	at com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:108)
	at com.github.benmanes.caffeine.cache.LocalManualCache.get(LocalManualCache.java:62)
	at org.apache.iceberg.CachingCatalog$CachingTableBuilder.create(CachingCatalog.java:257)
	at org.apache.iceberg.spark.SparkCatalog.createTable(SparkCatalog.java:164)
	at org.apache.iceberg.spark.SparkCatalog.createTable(SparkCatalog.java:102)
	at org.apache.spark.sql.execution.datasources.v2.CreateTableExec.run(CreateTableExec.scala:41)
	at org.apache.spark.sql.execution.datasources.v2.V2CommandExec.result$lzycompute(V2CommandExec.scala:40)
	at org.apache.spark.sql.execution.datasources.v2.V2CommandExec.result(V2CommandExec.scala:40)
	at org.apache.spark.sql.execution.datasources.v2.V2CommandExec.executeCollect(V2CommandExec.scala:46)
	at org.apache.spark.sql.Dataset.$anonfun$logicalPlan$1(Dataset.scala:228)
	at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:3687)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$5(SQLExecution.scala:103)
	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:163)
	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:90)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:775)
	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:3685)
	at org.apache.spark.sql.Dataset.(Dataset.scala:228)
	at org.apache.spark.sql.Dataset$.$anonfun$ofRows$2(Dataset.scala:99)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:775)
	at org.apache.spark.sql.Dataset$.ofRows(Dataset.scala:96)
	at org.apache.spark.sql.SparkSession.$anonfun$sql$1(SparkSession.scala:618)
	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:775)
	at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:613)
	at com.netease.arctic.server.terminal.local.LocalTerminalSession.executeStatement(LocalTerminalSession.java:78)
	at com.netease.arctic.server.terminal.TerminalSessionContext$ExecutionTask.executeStatement(TerminalSessionContext.java:281)
	at com.netease.arctic.server.terminal.TerminalSessionContext$ExecutionTask.execute(TerminalSessionContext.java:245)
	at com.netease.arctic.server.terminal.TerminalSessionContext$ExecutionTask.lambda$get$0(TerminalSessionContext.java:206)
	at com.netease.arctic.table.TableMetaStore.call(TableMetaStore.java:234)
	at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:207)
	at com.netease.arctic.server.terminal.TerminalSessionContext$ExecutionTask.get(TerminalSessionContext.java:196)
	at com.netease.arctic.server.terminal.TerminalSessionContext$ExecutionTask.get(TerminalSessionContext.java:171)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1604)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by wangtaohz**
## Why are the changes needed?

Close #2192.

## Brief change log

- put S3 related properties to catalog properties

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / **no**)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

**Comment by wangtaohz** [comment]
Local test result
image

=================================================

image

=================================================

image

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/CatalogController.java [hunk 1]
```diff
@@ -176,8 +178,8 @@ public void getCatalogTypeList(Context ctx) {
   }
 
   /** Convert server auth config to metaAuthConfig */
-  private Map<String, String> authConvertFromServerToMeta(
-      Map<String, String> serverAuthConfig, CatalogMeta oldCatalogMeta) {
+  private void authConvertFromServerToMeta(
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/CatalogController.java:181]: Maybe change the method name to `fillAuthConfigs` or `fillAuthConfigs2CatalogMeta` as you have changed the behavior and return type of this method.

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/CatalogController.java [hunk 2]
```diff
@@ -216,17 +218,26 @@ private Map<String, String> authConvertFromServerToMeta(
...
+            AUTH_CONFIGS_KEY_SECRET_KEY,
+            S3FileIOProperties.SECRET_ACCESS_KEY);
         break;
     }
-    return metaAuthConfig;
+    catalogMeta.setAuthConfigs(metaAuthConfig);
   }
 
   /** Convert meta auth config to server auth config DTO */
   private Map<String, Object> authConvertFromMetaToServer(
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/CatalogController.java:237]: `pareAuthConfigsFromCatalogMeta` may be a proper method name.

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/CatalogController.java [hunk 3]
```diff
@@ -254,17 +265,28 @@ private Map<String, Object> authConvertFromMetaToServer(
...
+            S3FileIOProperties.SECRET_ACCESS_KEY,
+            AUTH_CONFIGS_KEY_SECRET_KEY);
+        catalogMeta.getCatalogProperties().remove(S3FileIOProperties.SECRET_ACCESS_KEY);
         break;
     }
 
     return serverAuthConfig;
   }
 
   private Map<String, Object> storageConvertFromMetaToServer(
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/CatalogController.java:286]: `pareStorageConfigsFromCatalogMeta` may be a proper method name.

**Review by zhoujinsong — COMMENTED** [review]
I give some suggestion about the method name, please take another look.

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/CatalogController.java (hunk 2 above)
Comment (wangtaohz) [reply ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/CatalogController.java:237]: I noticed that we should not remove the auth properties from the catalog properties of `CatalogMeta`, which will  destroy the object. So `extractAuthConfigsFromCatalogMeta` may be better.

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/CatalogController.java (hunk 1 above)
Comment (wangtaohz) [reply ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/CatalogController.java]: I rename it to `fillAuthConfigs2CatalogMeta`

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/CatalogController.java (hunk 2 above)
Comment (zhoujinsong) [reply ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/CatalogController.java]: Sorry for my spelling mistake. I originally meant `parseAuthConfigsFromCatalogMeta`, but `extractAuthConfigsFromCatalogMeta` looks good to me too.

Comment (wangtaohz) [reply ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/CatalogController.java]: Now, I remove the auth config from properties of `CatalogSettingInfo` and `CatalogRegisterInfo` during `removeHiddenProperties`.

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/CatalogController.java (hunk 3 above)
Comment (wangtaohz) [reply ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/CatalogController.java]: I rename it to `extractStorageConfigsFromCatalogMeta`

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/CatalogController.java [hunk 4]
```diff
@@ -175,9 +197,9 @@ public void getCatalogTypeList(Context ctx) {
     ctx.json(OkResponse.of(catalogTypes));
   }
 
-  /** Convert server auth config to metaAuthConfig */
-  private Map<String, String> authConvertFromServerToMeta(
-      Map<String, String> serverAuthConfig, CatalogMeta oldCatalogMeta) {
+  /** fill server auth config to CatalogMeta */
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/CatalogController.java:200]: The comment here became meaningless to me as it is basically copied from the method name.

Comment (wangtaohz) [reply ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/CatalogController.java]: I removed it.

**Comment by wangtaohz** [comment]
When trying to update a hidden property (auth type is AK/SK)
image

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**